CareKitStore/CareKitStoreTests/OCKStore/TestStore+Outcomes.swift

```diff
@@ -227,14 +227,10 @@ class TestStoreOutcomes: XCTestCase {
         outcome = try store.addOutcomeAndWait(outcome)
 
         var query = OCKOutcomeQuery(for: Date())
-        query.tags = ["123"]
-
-        let fetched = try store.fetchOutcomesAndWait(query: query).first
-        var query2 = OCKOutcomeQuery(for: Date())
-        query2.uuids = [fetched!.uuid!]
+        query.uuids = [ outcome.uuid!]
```

Comment:
Looking good! The last change I'll ask you to make is to change `outcome.uuid!` to `try outcome.getUUID()`. If you look at some of the other tests, you'll see that we try to prefer using throwing getters over force unwraps in the context of unit tests.

Our reasoning is that should something happen to go wrong, we want the test to fail without crashing so that we can pinpoint the regression. If a force unwrap crashes the test runner, we won't be able to tell where the test failed.

Look around for a test extension on `OCKTask` that gives it the `func getUUID() throws -> UUID` method we use in tests and add another similar method for `OCKOutcome`, then employ that here.

Reply:
Added, but question. 

I added the getUUID extension on `OCKObjectCompatible` because the other versioned objects seemed to be getting it from `OCKVersionedObjectCompatible`. If this is okay, it would seem the extension `OCKVersionedObjectCompatible` can be removed because all versioned objects are `OCKObjectCompatible`.

I believe this enables getUUID on OCKNote and OCKValue and the like as well, which I'm not sure if that should be the case

Reply:
Yeah, you're 100% right. I think the right route is to just change the existing extension to be on `OCKObjectCompatible`. Previous only versioned objects had uuids, but now that outcomes do too, it makes sense to expand that scope of that extension!

Reply:
I'm sure it will pass, only thing now is a slight naming convention.

So in files in `buildPredicate` in versioned object files like OCKStore+CarePlans.swift, they use `versionPredicate` instead of `objectPredicate`. It still works of course:

```swift
if !query.uuids.isEmpty {
            let versionPredicate = NSPredicate(format: "%K IN %@", #keyPath(OCKCDVersionedObject.uuid), query.uuids)
            predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [predicate, versionPredicate])
        }
```

Should I change them or leave them?

Reply:
Might as well change them while we're at it!

Reply:
Thanks for boy-scouting!

Reply:
I'm think I may have found a bug in OCKStore+CarePlans.swift while changing these:

```
if !query.patientUUIDs.isEmpty {
            let objectPredicate = NSPredicate(format: "%K IN %@", #keyPath(OCKCDCarePlan.patient), query.patientUUIDs)
            predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [predicate, objectPredicate])
        }
```

I think that keypunch should be `OCKCDCarePlan.patient.uuid` instead of `OCKCDCarePlan.patient`

Reply:
Another one in contacts. These are probably here because of the conversion to UUIDs:

```
if !query.uuids.isEmpty {
            let versionPredicate = NSPredicate(format: "%K IN %@", #keyPath(OCKCDVersionedObject.id), query.uuids)
            predicate = NSCompoundPredicate(andPredicateWithSubpredicates: [predicate, versionPredicate])
        }
```

That should now be  `OCKCDObject.uuid `